### PR TITLE
Feature/save workflow

### PR DESCRIPTION
**Pull request by Aerendel**
For #83 

- Add save workflow button and save as dialog (note: used QT dialog instead of native OS for a more consistent experience. MacOSX doesn't honor file type filters for example when using the native dialog)
- Save current workflow as json with current input parameters

### CODE REVIEW

**Review by yrkim98 — COMMENTED** [review]
All the changes look good to me- I want to test saving and loading a workflow first before I approve! Will do that later this afternoon.

File: napari_aicssegmentation/view/workflow_steps_view.py [hunk 1]
```diff
@@ -200,10 +210,21 @@ def _btn_close_keep_clicked(self, checked: bool):
...
-        all_parameter_inputs = [w.get_parameter_inputs() for w in workflow_step_widgets]
+        all_parameter_inputs = [w.get_parameter_inputs() for w in self._get_workflow_step_widgets()]
         self._controller.run_all(all_parameter_inputs)
 
     def _btn_run_all_cancel_clicked(self, checked: bool):
         self.btn_run_all.setText("Canceling...")
         self._controller.cancel_run_all()
+
+    def _btn_save_workflow_clicked(self, checked: bool):
+        file_path, _ = QFileDialog.getSaveFileName(
```
Comment (schoinh) [inline napari_aicssegmentation/view/workflow_steps_view.py:221]: [nit] I had to look up what this `_` variable was -- if it's a string containing filters, maybe call it `filters` instead or if we're never going to use `filters` then not destructure the tuple that's returned? OR at the very least add a comment what `_` is :)

**Review by schoinh — APPROVED** [review]
I didn't pull the branch to test but looks good to me :)

File: napari_aicssegmentation/view/workflow_steps_view.py (hunk 1 above)
Comment (Aerendel) [reply napari_aicssegmentation/view/workflow_steps_view.py:221]: @schoinh in Python, by convention, the single underscore is used to unpack values you want to ignore when using tuples or loops. See the first part of https://betterprogramming.pub/the-special-meaning-of-underscores-in-python-9ceaaeb41007

Comment (schoinh) [reply napari_aicssegmentation/view/workflow_steps_view.py:221]: Ohh! Learned something new, thanks!

**Pull request merged**